pypesto/objective/amici_prediction.py

```diff
@@ -0,0 +1,439 @@
+            max_num_conditions: int = 0,
+    ) -> PredictionResult:
+        """
+        Method to simulate a model for a certain prediction function.
+        This method relies on the AmiciObjective, which is underlying, but
+        allows the user to apply any post-processing of the results and the
+        sensitivities.
+
+        Parameters
+        ----------
+        x:
+            The parameters for which to evaluate the prediction function.
+        sensi_orders:
+            Specifies which sensitivities to compute, e.g. (0,1) -> fval, grad.
+        mode:
+            Whether to compute function values or residuals.
+        output_file:
+            Path to an output file.
+        max_num_conditions:
+            In some cases, we don't want to compute all predictions at once
+            when calling the prediction function, as this might not fit into
+            the memory for large datasets and models.
+            Here, the user can specify a maximum number of conditions, which
+            should be simulated at a time. Default is self.max_num_conditions.
+        output_format:
+            Either 'csv', 'h5'. If an output file is specified, this routine
+            will return a csv file, created from a DataFrame, or an h5 file,
+            created from a dict.
+
+        Returns
+        -------
+        outputs:
+            List of postprocessed outputs (ndarrays)
+        outputs_sensi:
+            List of sensitivities of postprocessed outputs (ndarrays)
+        """
+        # sanity check for output
+        if 2 in sensi_orders:
+            raise Exception('Prediction simulation does currently not support '
+                            'second order output.')
+
+        # prepare for storing the results
+        amici_y = []
+        amici_sy = []
+        amici_t = []
+
+        # simulate the model and get the output
+        self._get_model_outputs(amici_y, amici_sy, amici_t, x,
+                                sensi_orders, mode)
+
+        # postprocess
+        if self.post_processing is not None:
+            outputs = self.post_processing(amici_y)
+        else:
+            outputs = amici_y
+        if self.post_processing_sensi is not None:
+            outputs_sensi = self.post_processing_sensi(amici_y, amici_sy)
+        else:
+            outputs_sensi = amici_sy
+        if self.post_processing_timepoints is not None:
+            timepoints = self.post_processing_timepoints(amici_t)
+        else:
+            timepoints = amici_t
+
+        condition_results = []
+        for i_cond in range(len(timepoints)):
+            result = {'timepoints': timepoints[i_cond],
+                      'observables': self.observables}
+            if outputs:
+                result['output'] = outputs[i_cond]
+            if outputs_sensi:
+                result['output_sensi'] = outputs_sensi[i_cond]
+
+            condition_results.append(result)
+
+        results = PredictionResult(condition_results)
```

Comment:
Of course could. Again, that's a design question, I think. My intention was: Keeping the objects which are moved around and kept in the memory small. I'll be using this for a model with >1k state, >4k parameters, roughly 2k conditions, but only few observables. Simulating this and keeping all rdata-objetcs in the memory kills my computer, as this exceeds the available RAM. So, my idea is: Keep things as small as possible.

One could just as well say: If you're interested in your states, add them to the observables. ;) This brings other problems, such as longer compile time, however... So, I'm open to any discussion here. I had a reason of designing it that way. If there are better reasons to do it another way, I'm open for it...

Reply:
Maybe a non-empty list could be provided with any combination of `{'states', 'observables'}`, such that users can describe what they would like in the results object? I think states would be good, but then as an option to avoid memory issues.

But OK as is for this PR.